### DISCUSSION TITLE
Change led color on touch

### DISCUSSION THREAD

**Discussion by Fabaderheld** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#1] [Q&A] [answered]
Hi,

I have played around with TX Ultimate with ESPHome, you wanted to know if I change the color of the LED from red to blue on touch. 

I think I found the part of the configuration but somehow can't add it to my config (I'm using the minimal config at the moment)

~~~
  - id: touch_led_marker
    mode: restart
    parameters:
      state: int
      pos: int
    then:
      - if:
          condition:
            lambda: "return state==1;"
          then:
            # RELEASE
            - light.turn_off: leds

          else:
            - if:
                condition:
                  lambda: "return state==2;"
                then:
                  # TOUCH
                  - light.turn_on:
                      id: leds
                      red: 1
                      green: 0
                      blue: 0
                  - delay: 1s
                  - light.turn_off: leds
~~~

Any help is appreciated

Faba

**Comment by danielscheidler** [comment, marked as answer]
You can use the [tx_ultimate_local.yaml](https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/blob/main/tx_ultimate_local.yaml) instead of the minimal config.
Then you can change all actions/colors on your need.

First you can try only to update your device. I've done an update today with new visualizations for touch, show relay-state and a new nightlight feature. The new touch color is blue already.

**Reply by Fabaderheld** [reply]
Thanks Daniel,

Yeah that's what I've figured out by myself. But I'll have a look a the new version.

Regards

Faba

**Reply by danielscheidler** [reply]
I've added some new substitutions to set all colors, brightness and effects.

```
  touch_brightness: "1"
  touch_color: "{0,100,100}"
  touch_effect: "Scan"

  long_press_brightness: "1"
  long_press_color: "{100,0,0}"
  long_press_effect: ""

  multi_touch_brightness: "1"
  multi_touch_color: "{0,0,0}"
  multi_touch_effect: "Rainbow"

  swipe_left_brightness: "1"
  swipe_left_color: "{0,100,0}"
  swipe_left_effect: ""

  swipe_right_brightness: "1"
  swipe_right_color: "{100,0,70}"
  swipe_right_effect: ""
```

**Reply by Fabaderheld** [reply]
Damn you are fast! Maybe I should wait some time until I flash all of my TX Ultimates.

**Reply by danielscheidler** [reply]
Why you'll wait? If you use the minimal version as a basis and only install your extensions, you just have to flashing again without changing anything. (Except possibly set the new substitutions. But that is only necessary if you want to change the standard behavior)
If you have any further questions, wishes or ideas, please let me know.

**Reply by Fabaderheld** [reply]
I'm not that fit with ESPHome and at our new House I'll have 30+ switches to flash and if I change the config once a week it gets messy :D

Just a general Question, where would you configure settings in Home Assistant or ESPHome itself? I can also switch to the discord channel if it fits better there. 

Schöne Grüße aus Wien

Faba

**Reply by danielscheidler** [reply]
Ah na dann muss ich mir ja keinen mit Englisch abbrechen 😄
Also wohin die Logik gehört kann man immer drüber streiten. Ich halte es eigtl so, das alles was das Gerät selber betrifft auch auf dem Gerät, also in esphome, erfolgt. 
Alles was sich auf andere Geräte bezieht regel ich zentral in Home Assistant.
Also LEDs und Relais im Schalter gehört für mich ins esphome. Wenn ich mit Wischgesten aber zb die gardienen steuern will, wäre das ne Automatisierung im HA

**Reply by Fabaderheld** [reply]
Englisch is kein Problem ;), dachte es ist besser wenn jemand das repo finden sollte wo alles auf englisch ist. 

Ja so ungefähr hab ich es jetzt auch gelöst, versuch gerade eine Notifizierung für die Türklingel zu bauen, wenn der Sound ordentlich funktionieren würde wäre es der hammer. ;)

schönen Abend noch

**Comment by Fabaderheld** [comment]
Is there a way to pay you a coffee or beer?

**Reply by danielscheidler** [reply]
Yes. All options to support me, you'll find here:
Https://smarthomeyourself.de/support

**Comment by danielscheidler** [comment]
Hab die Anleitung ja auch 2sprachig.
Ja mit dem Ton ist schade. Hatte es mal versucht aber geht nur mit weiterer
custom component. Und da es eh als Anpassung in esphome schon in Arbeit ist
macht das glaub ich wenig Sinn.
Sobald der Patch draussen ist integrier ich das auf jeden Fall auch.
Mit dem nächsten esphome Update muss eh die i2c audio Komponente angepasst
werden. Die haben die yaml Syntax geändert.
Aber ist schon vorbereitet.

Fabian Sasse ***@***.***> schrieb am Do., 8. Juni 2023, 16:47:
